Makefile.core.mk

```diff
@@ -296,7 +296,7 @@ ifeq ($(BUILD_ALL),true)
 $(ISTIO_OUT_LINUX)/$(shell basename $(1)): build-linux
 else
 $(ISTIO_OUT_LINUX)/$(shell basename $(1)): $(ISTIO_OUT_LINUX)
-	STATIC=0 GOOS=linux GOARCH=amd64 LDFLAGS='-extldflags -static -s -w' common/scripts/gobuild.sh $(ISTIO_OUT_LINUX)/ $(1)
+	STATIC=0 GOOS=linux GOARCH=$(GOARCH_LOCAL) LDFLAGS='-extldflags -static -s -w' common/scripts/gobuild.sh $(ISTIO_OUT_LINUX)/ $(1)
```

Comment:
This isn't correct. The build system will need significant modification to support arm. Keep in mind the development environment cannot see any regression. This would break the 60 or so test cases for the developer system (that just merged btw..) to enable arm. If you need more information, feel free to check out the developer wiki: https://github.com/istio/istio/wiki